File: lib/predict/classify/classification_result.dart
/// A result of an image classification.
class ClassificationResult {
  /// Creates a [ClassificationResult].
  ClassificationResult({
    required this.index,
    required this.label,
    required this.confidence,
  });

  /// Creates a [ClassificationResult] from a [json] object.
  factory ClassificationResult.fromJson(Map<String, dynamic> json) =>
      ClassificationResult(
        index: json['index'] as int,
        label: json['label'] as String,
        confidence: json['confidence'] as double,
      );

  /// The index of the label.
  final int index;

  /// The label of the classification.
  final String label;

  /// The confidence of the classification.
  final double confidence;
}

File: android/src/main/java/com/ultralytics/ultralytics_yolo/predict/classify/ClassificationResult.java
// Ultralytics 🚀 AGPL-3.0 License - https://ultralytics.com/license

package com.ultralytics.ultralytics_yolo.predict.classify;

public class ClassificationResult {
    public final String label;
    public final float confidence;
    public final int index;

    public ClassificationResult(String label, int index, float confidence) {
        this.label = label;
        this.index = index;
        this.confidence = confidence;
    }
}
